shop/src/main/java/ro/msg/learning/shop/service/impl/ProductCategoryServiceImpl.java

```diff
@@ -0,0 +1,22 @@
+package ro.msg.learning.shop.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.model.ProductCategory;
+import ro.msg.learning.shop.repository.ProductCategoryRepository;
+import ro.msg.learning.shop.service.ProductCategoryService;
+
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class ProductCategoryServiceImpl implements ProductCategoryService {
+
+    private final ProductCategoryRepository productCategoryRepository;
+
+
+    @Override
+    public Optional<ProductCategory> findByName(String name) {
+        return productCategoryRepository.getProductCategoryByName(name);
```

Comment:
Try adding some null checks for these values. This can also be achieved by using Spring Validators. 
https://www.baeldung.com/spring-boot-bean-validation